### DISCUSSION TITLE
Can more WebRTC extensions be added in the SDP negotiation?

### DISCUSSION THREAD

**Discussion by MarcoMayCry** [paullouisageneau/libdatachannel#926] [Q&A]
Recently, OBS has used this library to implement WebRTC streaming (WHIP). However, I noticed that there are no `rtx` line and `rrtr` line in the SDP negotiation. Does this library support these two extensions (RTX & RTCP Extended Reports (XR))? Or, how can I enable these two extensions in the SDP negotiation?

![gpp6KHm5G6KlbGExzSOESgz0j5PN6dKyrEVpd7wvxIYXs7dd51814zWJco0E4ew8-qi-ye-wei-xin-jie-tu-_72ed66d6-bbae-43a9-a157-941cd55ffcf4](https://github.com/paullouisageneau/libdatachannel/assets/79976248/d7a534c0-cbc0-4bf6-8d98-d34a7e75d17f)

The RRTR line will enable RTCP Extended Reports (XR) so that the server can update the RTT in real time, thereby improving the efficiency of strategies such as NACK.

**Comment by paullouisageneau** [comment]
These extensions are not supported for now, so you can't enable them via SDP.

**Reply by MarcoMayCry** [reply]
Thank you very much for your reply. Are there any plans to support these extensions in the near future? I noticed that Chrome should have already supported these extensions.

**Comment by vipcxj** [comment]
@paullouisageneau I found TODO about why rtx is not implemented, something to do with firefox bugs. But I also found a lot of api related to rtx. including the api to write rtx lines to sdp. but then I looked at the nackhandler source and found that the ssrc of the nack package didn't change, which means that it doesn't use rtx, so now rtx is still unimplemented, is that right?
